src/mcfret.c

```diff
@@ -0,0 +1,72 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <string.h>
+#include <time.h>
+#include <fftw3.h>
+#include "omp.h"
+#include "types.h"
+#include "NISE_subs.h"
+#include "mcfret.h"
+#include "project.h"
+
+/* Distribute the MCFRET calculations */
+void mcfret(t_non *non){
+    /* Define variables and arrays */
+    /*Number of molecules*/
+    int nn2;
+    /* Response functions for emission and absorption: real and imaginary part*/
+    float *re_S_Abs,*im_S_Abs;
+    float *re_S_Emi,*im_S_Emi;
+    /*Hamiltonian of the whole system */
+    float *Hamil_i_e;
+
+
+    /*Allocate memory for all the variables*/
+    /*Allocate memory for the response functions*/
+    nn2=non->singles*(non->singles+1)/2;
+    re_S_Abs=(float *)calloc(nn2*nn2*non->tmax,sizeof(float));
```

Comment:
This should be non->singles*non->singles instead of nn2*nn2 as the absorption matrix has the dimension of the number of sites squared